### DISCUSSION TITLE
How to find and replace text in the entire website?

### DISCUSSION THREAD

**Discussion by markushuebner** [datenstrom/community#1001] [Ask a question] [answered]
Hi all,

is there a plugin or tool to bulk replace content within md-files? E.g. if I mention "Petra Lastname" in several articles but her name changes to "Petra Newlastname", I would have to search and replace the occurences manually.

Another example: If I change the structure of the website, several internal links won't work anymore. Structure before changes

_000-startpage
001-topic-a
002-topic-b/001-subtopic-c_

working link: _/topic-b/subtopic-c/_

If I restructure to 

_000-startpage
001-topic-a/001-subtopic-c
002-topic-b_

that link of course won't work anymore. It would be very handy if one can search and replace the old link
_/topic-b/subtopic-c/_
with the new link
_/topic-a/subtopic-c/_
throughout all md-files automatically.

Thanks a lot and greetings,

Markus

**Comment by GiovanniSalmeri** [comment, marked as answer]
> If I change the structure of the website, several internal links won't work anymore

I wrote [Link](https://github.com/GiovanniSalmeri/yellow-link) in order to prevent this problem in my sites. Hope it will be useful to you!

As regards the general problem of bulk replacing: it would be possible to write an extension for this purpose. But if you have a local copy of your website in your computer, it is rather easy to use the common editing programs (in Linux I use `sed` or a combination of  `grep` and `gvim`, but every operative system has its own tools). I find this possibility one of the best aspects of Yellow!

**Reply by annaesvensson** [reply]
The [check extension](https://github.com/annaesvensson/yellow-check) does also find broken links, maybe it can be another useful tool.

**Reply by markushuebner** [reply]
So, I spend some time to check out the helloworld extension and managed it to "write" my own one to show some styled phrases and JS-magic directly and via including content of a php-file in another folder.

My aim was/is, to implement my full php-script "as is" via an extension. But it does not seem to be possible, at least I do not get it how to solve that problem. I would rather leave my standalone application as is, located in an htaccess-protected folder.

Perhaps it is possible that one of you dev-cracks (I am not one) might have a look at my little, quite handy tool?

Thank you and greetings,

Markus

**Reply by annaesvensson** [reply]
Want to show your source code to other developers? Create a repository and upload your files to GitHub.

**Comment by markushuebner** [comment]
Thank you, Giovanni.

I'll have look at "Link". I wrote a standalone search and replace tool some years ago and perhaps it might be a good addition for yellow, regarding the concept of free phrasing the search and replace strings. But first, I have to look closer at how to create an extension for yellow and contribute correctly via github.
